src/veritysetup/veritysetup.c

```diff
@@ -109,11 +121,63 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "hash-offset="))) {
+                else if ((val = startswith(word, "no-superblock="))) {
+
+                        r = parse_boolean(val);
```

Comment:
ah, you already parse a rhs boolean, good – but in that case the lhs should not be negated, i.e. please remove the "no-" logic, invert the variable